### PR TITLE
Various camera improvements

### PR DESCRIPTION
**Pull request by julianoes**
As part of testing against the camera-manager, various issues are coming up.

### CODE REVIEW

File: plugins/camera/camera_impl.cpp [hunk 1]
```diff
@@ -107,13 +106,31 @@ void CameraImpl::disable()
...
+
 MAVLinkCommands::CommandLong CameraImpl::make_command_request_camera_info()
 {
     MAVLinkCommands::CommandLong command_camera_info{};
 
     command_camera_info.command = MAV_CMD_REQUEST_CAMERA_INFORMATION;
     command_camera_info.params.param1 = 1.0f; // Request it
-    command_camera_info.target_component_id = MAV_COMP_ID_CAMERA;
+    command_camera_info.target_component_id = _component_id;
+    ;
```
Comment (shakthi-prashanth-m) [inline plugins/camera/camera_impl.cpp]: Maybe typo

File: plugins/camera/camera_impl.cpp [hunk 2]
```diff
@@ -107,13 +106,31 @@ void CameraImpl::disable()
     _parent->remove_call_every(_flight_information_call_every_cookie);
 }
 
+Camera::Result CameraImpl::select_camera(unsigned id)
+{
+    if (id > 6) {
```
Comment (shakthi-prashanth-m) [inline plugins/camera/camera_impl.cpp]: We support Ids's 0 to 5 right ? If so, it should be 
```suggestion
    const size_t MAX_SUPPORTED_IDS = 5; // inside class
    if (id > MAX_SUPPORTED_IDS) {
```

File: plugins/camera/camera_impl.cpp (hunk 1 above)
Comment (julianoes) [reply plugins/camera/camera_impl.cpp]: Thanks!

File: plugins/camera/camera_impl.cpp (hunk 2 above)
Comment (julianoes) [reply plugins/camera/camera_impl.cpp]: Good catch! I'll fix it.

File: plugins/camera/include/plugins/camera/camera.h [hunk 3]
```diff
@@ -65,6 +66,22 @@ class Camera : public PluginBase {
      */
     static std::string result_str(Result result);
 
+    /**
+     * @brief Select camera to interact with.
+     *
+     * @param id The camera ID from 0 to 5.
+     *
+     * Currently supported are IDs 0 to 5 mapping to MAVLink component IDs
```
Comment (shakthi-prashanth-m) [inline plugins/camera/include/plugins/camera/camera.h]: Is it necessary to mention MAVLink ?

Comment (julianoes) [reply plugins/camera/include/plugins/camera/camera.h]: I think for now this can be helpful.

File: plugins/camera/camera_impl.cpp [hunk 4]
```diff
@@ -87,24 +92,76 @@ void CameraImpl::deinit()
...
+            _camera_found = true;
+            manual_enable();
+        }
+    }
 }
 
 void CameraImpl::enable()
+{
+    // FIXME: We check for the connection status manually because
+    // we're not interested int the connection state of the autopilot
```
Comment (JonasVautherin) [inline plugins/camera/camera_impl.cpp]:
```suggestion
    // we're not interested in the connection state of the autopilot
```

File: plugins/camera/camera_impl.cpp [hunk 5]
```diff
@@ -87,24 +92,76 @@ void CameraImpl::deinit()
...
+    request_flight_information();
+
     _parent->add_call_every(
         [this]() { request_flight_information(); }, 10.0, &_flight_information_call_every_cookie);
 }
 
 void CameraImpl::disable()
+{
+    // FIXME: We check for the connection status manually because
+    // we're not interested int the connection state of the autopilot
```
Comment (JonasVautherin) [inline plugins/camera/camera_impl.cpp]:
```suggestion
    // we're not interested in the connection state of the autopilot
```

File: plugins/camera/camera_impl.cpp [hunk 6]
```diff
@@ -87,24 +92,76 @@ void CameraImpl::deinit()
...
 {
     invalidate_params();
     _parent->remove_call_every(_flight_information_call_every_cookie);
+
+    _camera_found = false;
+}
+
+Camera::Result CameraImpl::select_camera(unsigned id)
+{
+    static constexpr std::size_t MAX_SUPPORTED_ID = 5;
```
Comment (JonasVautherin) [inline plugins/camera/camera_impl.cpp]: Why is that?

File: plugins/camera/camera_impl.cpp [hunk 7]
```diff
@@ -87,24 +92,76 @@ void CameraImpl::deinit()
...
+
+void CameraImpl::manual_disable()
 {
     invalidate_params();
     _parent->remove_call_every(_flight_information_call_every_cookie);
+
+    _camera_found = false;
+}
+
+Camera::Result CameraImpl::select_camera(unsigned id)
```
Comment (JonasVautherin) [inline plugins/camera/camera_impl.cpp]: Is the idea the following?

1. We instantiate a `camera` object
2. We select the desired camera
3. If needed, we can instantiate another `camera` object and set it to another camera

Comment (julianoes) [reply plugins/camera/camera_impl.cpp]: I guess so. I agree it's ugly.

File: plugins/camera/camera_impl.cpp (hunk 6 above)
Comment (julianoes) [reply plugins/camera/camera_impl.cpp]: 'Cause MAVLink definition.

Comment (JonasVautherin) [reply plugins/camera/camera_impl.cpp]: Fair enough :D

**Pull request merged**